server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1122,6 +1116,11 @@ public static Collection<String> resolve(Context context, List<String> expressio
                 if (expressions.size() == 1 && expressions.get(0).equals(Metadata.ALL)) {
                     return List.of();
                 } else {
+                    if (context.getOptions().ignoreUnavailable() == false) {
+                        for (String expression : expressions) {
+                            ensureAliasOrIndexExists(context, expression);
+                        }
+                    }
```

Comment:
This is the logic to handle the ignore unavailable resources when wildcards are not expanded.